src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -136,7 +141,67 @@ async function processPgSQL(queryArray: any[], queryType: string, connection: IC
   }
 }
 
-function processGraphQL(queryArray: any[], connection: object) {
-  // TO-DO -- For now just return a value that makes test work
-  return 0
+async function processGraphQL(queryArray: any[], connection: IGraphQLConnection) {
+  const query = queryArray[0]
+  const variableNames = queryArray[1]
+  const variableNodes = queryArray.slice(2, queryArray.length - 1)
+  const returnNode = queryArray[queryArray.length - 1]
+
+  const variables = zipArraysToObject(variableNames, variableNodes)
+
+  const data = await graphQLquery(query, variables, connection)
+
+  const selectedNode = extractNode(data, returnNode)
+
+  if (Array.isArray(selectedNode)) return selectedNode.map((item) => simplifyObject(item))
+  else return simplifyObject(selectedNode)
+}
+
+// Build an object from an array of field names and an array of values
+function zipArraysToObject(variableNames: string[], variableValues: any[]) {
+  const createdObject: BasicObject = {}
+  variableNames.map((name, index) => {
+    createdObject[name] = variableValues[index]
+  })
+  return createdObject
+}
+
+// Return a specific node (e.g. application.name) from a nested Object
+function extractNode(
+  data: BasicObject,
+  node: string
+): BasicObject | string | number | boolean | BasicObject[] {
+  const returnNodeArray = node.split('.')
+  return extractNodeWithArray(data, returnNodeArray)
+
+  function extractNodeWithArray(
+    data: BasicObject,
+    nodeArray: string[]
+  ): BasicObject | string | number | boolean | BasicObject[] {
+    if (nodeArray.length === 1) return data[nodeArray[0]]
+    else return extractNodeWithArray(data[nodeArray[0]], nodeArray.slice(1))
+  }
+}
+
+// If Object has only 1 field, return just the value of that field,
+// else return the whole object.
+function simplifyObject(item: number | string | boolean | BasicObject) {
+  if (typeof item === 'object' && Object.keys(item).length === 1) return Object.values(item)[0]
+  else return item
+}
+
+async function graphQLquery(query: string, variables: object, connection: IGraphQLConnection) {
+  const queryResult = await connection.fetch(connection.endpoint, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Accept: 'application/json',
+    },
+    body: JSON.stringify({
+      query: query,
+      variables: variables,
+    }),
+  })
+  const data = await queryResult.json()
+  return data.data
```

Comment:
In this case the break-down just make it more difficult to see what the graphQL does, I suggest combining this with the client function `async function processGraphQL` if possible.

Reply:
Hmm, I thought pulling all the boilerplate `fetch` stuff out of the main function made it much cleaner.

Does this line not give a clear indication of what's happening?
`const data = await graphQLquery(query, variables, connection)`

Reply:
Yes. It's good for now. No worries. 

I like the changes you made for the remaining suggestion too.